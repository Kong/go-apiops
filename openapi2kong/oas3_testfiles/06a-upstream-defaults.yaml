# Some of the generated fields for upstreams should be overridable
# using the defaults.
# These are the order in which we should take the values:
# - upstream.targets: x-kong-upstream-defaults -> list of targets gotten from servers block -> default empty array

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: https://server1.com/anything

x-kong-upstream-defaults:
  targets:
    - target: 'my-server1.com:443'
    - target: 'my-server2.com:443'

paths:
  /path1:
    get:
      operationId: uses-doc-service
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
