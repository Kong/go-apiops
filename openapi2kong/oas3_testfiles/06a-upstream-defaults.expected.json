{
  "_format_version": "3.0",
  "services": [
    {
      "host": "Simple_API_overview.upstream",
      "id": "e3a95664-95d7-5b10-bc3d-11c9cc6c2b4e",
      "name": "Simple_API_overview",
      "path": "/anything",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "5f917263-5851-5850-923f-06a35814bb61",
          "methods": [
            "GET"
          ],
          "name": "Simple_API_overview-uses-doc-service",
          "paths": [
            "~/path1$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_06a-upstream-defaults.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_06a-upstream-defaults.yaml"
      ]
    }
  ],
  "upstreams": [
    {
      "id": "dca5eafe-311a-520f-87bc-0e695b01eab6",
      "name": "Simple_API_overview.upstream",
      "tags": [
        "OAS3_import",
        "OAS3file_06a-upstream-defaults.yaml"
      ],
      "targets": [
        {
          "tags": [
            "OAS3_import",
            "OAS3file_06a-upstream-defaults.yaml"
          ],
          "target": "my-server1.com:443"
        },
        {
          "tags": [
            "OAS3_import",
            "OAS3file_06a-upstream-defaults.yaml"
          ],
          "target": "my-server2.com:443"
        }
      ]
    }
  ]
}